src/key_variants/double_key.rs

```diff
@@ -98,18 +164,32 @@ pub struct Proof {
 }
 
 impl Proof {
```

Comment:
Just noting that the name `Proof` is poorly chosen here since it could be mistaken for a plonk proof. In the `single_key` module this is named `Signature` which in my opinion is much more fitting. This is out of scope of this PR though and need to be tackled in a new issue

Reply:
created #89